### PR TITLE
Update/v25.0.3

### PR DESCRIPTION
**Pull request by islandbitcoin**

### CODE REVIEW

File: scripts/services/migrations.ts [hunk 1]
```diff
@@ -29,27 +29,7 @@ export const migration: T.ExpectedExports.migration = compat.migrations
...
-                path: "start9/password.dat",
-              });
-              config["enable-tor"] = false;
-            }
-            return config;
-          },
-          false,
-          { version: "25.0.3", type: "down" },
-        ),
+        down: () => { throw new Error('Cannot downgrade') },
```
Comment (chrisguida) [inline scripts/services/migrations.ts:32]:
```suggestion
        down: () => { throw new Error('Downgrade prohibited') },
```

File: scripts/services/migrations.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ export const migration: T.ExpectedExports.migration = compat.migrations
               delete config.username;
               await effects.writeFile({
                 path: "start9/password.dat",
-                toWrite: config.password || "",
+                toWrite: "password: "+ config.password || "",
```
Comment (chrisguida) [inline scripts/services/migrations.ts:21]: What's the reason for this change? Seems it makes the file more complex

Comment (islandbitcoin) [reply scripts/services/migrations.ts:21]: necessary for `yq` to work

Comment (islandbitcoin) [reply scripts/services/migrations.ts:21]: I could prob just `cat` the file instead in the entry point...

File: docker_entrypoint.sh [hunk 3]
```diff
@@ -42,22 +42,27 @@ echo '    qr: false' >> /root/start9/stats.yaml
...
   # mkdir -p /etc/ssl/certs
-  cp /mnt/cert/rest.key.pem /etc/ssl/certs/key.pem
-  while ! [ -e /mnt/cert/rest.cert.pem ]; do
+  cp /mnt/cert/main.key.pem /etc/ssl/private/ssl-cert-snakeoil.key
+  while ! [ -e /mnt/cert/main.cert.pem ]; do
     echo "Waiting for system cert..."
     sleep 1
   done
-  cp /mnt/cert/rest.cert.pem /etc/ssl/certs/certificate.pem
+  cp /mnt/cert/main.cert.pem /etc/ssl/certs/ssl-cert-snakeoil.pem
```
Comment (chrisguida) [inline docker_entrypoint.sh:59]: is this really the name of the file??

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+    "deno.enable": true
+}
```
Comment (chrisguida) [inline .vscode/settings.json:3]: This probably shouldn't be checked in.

**Review by chrisguida — CHANGES_REQUESTED** [review]
See comment

File: docker_entrypoint.sh (hunk 3 above)
Comment (islandbitcoin) [reply docker_entrypoint.sh:59]: yup. @dr-bonez confirmed

File: .vscode/settings.json (hunk 4 above)
Comment (islandbitcoin) [reply .vscode/settings.json:3]: fixed

File: docker_entrypoint.sh [hunk 5]
```diff
@@ -105,4 +120,4 @@ fi
 
 trap _term SIGTERM
 
-wait -n $nextcloud_process
\ No newline at end of file
+wait -n $nextcloud_process
```
Comment (k0gen) [inline docker_entrypoint.sh:123]: I see nextcloud_process is referenced in lines 7 and 123 but not assigned anywhere. I guess line 80 would be a good candidate but tini is being used.

File: docker_entrypoint.sh [hunk 6]
```diff
@@ -9,48 +9,60 @@ _term() {
...
 LAN_ADDRESS=$(yq e '.lan-address' /root/start9/config.yaml)
-CONNECTION=$(yq e '.enable-tor' /root/start9/config.yaml)
 SERVICE_ADDRESS='nextcloud.embassy'
-NEXTCLOUD_ADMIN_USER=$(yq e '.username' /root/start9/config.yaml)
-NEXTCLOUD_ADMIN_PASSWORD=$(yq e '.password' /root/start9/config.yaml)
+NEXTCLOUD_ADMIN_USER='embassy'
 POSTGRES_DATADIR="/var/lib/postgresql/13"
 POSTGRES_CONFIG="/etc/postgresql/13"
 NEXTCLOUD_TRUSTED_DOMAINS="$TOR_ADDRESS $LAN_ADDRESS $SERVICE_ADDRESS"
 TRUSTED_PROXIES="$TOR_ADDRESS $LAN_ADDRESS $SERVICE_ADDRESS"
```
Comment (k0gen) [inline docker_entrypoint.sh:17]: I don't see NEXTCLOUD_TRUSTED_DOMAINS and TRUSTED_PROXIES being used internally. Verify if this is working, bash might be more forgiving but normally this should be exported.

**Review by k0gen — COMMENTED** [review]
process handling in docker_entrypoint.sh needs more attention and testing rest is looking good

File: scripts/services/migrations.ts [hunk 7]
```diff
@@ -1,7 +1,36 @@
...
+          false, // setting to needs config due to potential bug on service update
+          { version: "25.0.3.3", type: "up" },
+        ),
+        down: () => { throw new Error('Downgrade prohibited') },
+      },
     },
-    "25.0.2" 
-);
+    "25.0.3.3",
+  );
```
Comment (chrisguida) [inline scripts/services/migrations.ts:36]: What's this for?

**Review by chrisguida — APPROVED** [review]
Seems fine

File: docker_entrypoint.sh (hunk 5 above)
Comment (islandbitcoin) [reply docker_entrypoint.sh:123]: can you propose a solution to be merged?

File: docker_entrypoint.sh (hunk 6 above)
Comment (islandbitcoin) [reply docker_entrypoint.sh:17]: yes

```

2023-02-06T18:04:46-05:00  Nextcloud was successfully installed
2023-02-06T18:04:46-05:00  Setting trusted domains…
2023-02-06T18:04:47-05:00  System config value trusted_domains => 1 set to string f2hbleftsndxaitsfju72ynyrwzho4si3a5vutvkwvsktfhykbxmk5id.onion
2023-02-06T18:04:48-05:00  System config value trusted_domains => 2 set to string f2hbleftsndxaitsfju72ynyrwzho4si3a5vutvkwvsktfhykbxmk5id.local
2023-02-06T18:04:49-05:00  System config value trusted_domains => 3 set to string nextcloud.embassy
2023-02-06T18:04:49-05:00  Initializing finished
```

File: scripts/services/migrations.ts (hunk 7 above)
Comment (islandbitcoin) [reply scripts/services/migrations.ts:36]: my understanding is the migration.ts current version should keep in sync with the manifest version

**Review by kn0wmad — APPROVED** [review]
